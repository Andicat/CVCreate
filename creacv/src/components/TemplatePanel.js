import React from 'react';
import PropTypes from 'prop-types';
import TemplateGroup from './TemplateGroup';
import CSSTransitionGroup from 'react-transition-group';

class TemplatePanel extends React.PureComponent {

    static propTypes = {
        groups: PropTypes.array,
    };

    static defaultProps = {
        groups: [],
    };

    state = {
        activeMenuId: null,
    }

    selectMenu = (id) => {
        this.setState( {activeMenuId:(id===this.state.activeMenuId)?null:id} );
    }

    render () {
        var groupsCode = this.props.groups.map( (g,i) => {
            return <TemplateGroup key={i} id={i} data={g} active={(this.state.activeMenuId===i)?true:false} cbSelected={this.selectMenu}/>
            });

        return <ul className='template-panel__menu'>
                    <CSSTransitionGroup transitionName="example">
                        {groupsCode}
                    </CSSTransitionGroup>    
               </ul>
    }
}

export default TemplatePanel;

