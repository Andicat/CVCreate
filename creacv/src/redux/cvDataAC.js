const CV_BLOCK_ADD = 'CV_BLOCK_ADD';
const CV_BLOCK_DELETE = 'CV_BLOCK_DELETE';
const CV_BLOCK_MOVE = 'CV_BLOCK_MOVE';
const CV_BLOCK_ACTIVATE = 'CV_BLOCK_ACTIVATE';

const cvBlock_add = function(block) {
  return {
    type: CV_BLOCK_ADD,
    block:block,
  };
}

const cvBlock_delete = function(block) {
  return {
    type: CV_BLOCK_ADD,
    block:block,
  };
}

const cvBlock_move = function(block) {
  return {
    type: CV_BLOCK_MOVE,
    block:block,
  };
}

const cvBlock_activate = function(block) {
  return {
    type: CV_BLOCK_ACTIVATE,
    block:block,
  };
}

export {
  cvBlock_add, CV_BLOCK_ADD,
  cvBlock_delete, CV_BLOCK_DELETE,
  cvBlock_move, CV_BLOCK_MOVE,
  cvBlock_activate, CV_BLOCK_ACTIVATE,
}
