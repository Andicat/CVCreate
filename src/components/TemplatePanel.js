import React from 'react';
import PropTypes from 'prop-types';
import TemplateGroup from './TemplateGroup';

import {connect} from 'react-redux';

class TemplatePanel extends React.PureComponent {

    static propTypes = {
        transitionClass: PropTypes.string,
        templatesArr: PropTypes.array,
    };

    state = {
        activeMenuId: null,
    }

    selectMenu = (id) => {
        this.setState( {activeMenuId:(id===this.state.activeMenuId)?null:id} );
    }

    render () {
        //console.log('render template panel',this.props);
        var groupsCode = this.props.templatesArr.map( (g,i) => {
            return <TemplateGroup key={i} id={i} data={g} active={(this.state.activeMenuId===i)?true:false} cbSelected={this.selectMenu}/>
            });

        return <ul className={'template-panel__menu ' + this.props.transitionClass}>
                    {groupsCode}
               </ul>
    }
}

const mapStateToProps = function (state) {
    return {
        templatesArr: state.templatesData.templatesArr,
    };
};
  
export default connect(mapStateToProps)(TemplatePanel);
