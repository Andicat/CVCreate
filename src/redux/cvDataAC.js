const CV_BLOCK_ADD = 'CV_BLOCK_ADD';
const CV_BLOCK_DELETE = 'CV_BLOCK_DELETE';
const CV_BLOCK_MOVE = 'CV_BLOCK_MOVE';
const CV_BLOCK_RESIZE = 'CV_BLOCK_RESIZE';
const CV_BLOCK_ACTIVATE = 'CV_BLOCK_ACTIVATE';
const CV_BLOCK_ACTIVATE_MULTI = 'CV_BLOCK_ACTIVATE_MULTI';
const CV_BLOCK_COPY = 'CV_BLOCK_COPY';
const CV_BLOCK_SET_SIZE = 'CV_BLOCK_SET_SIZE';
const CV_ELEMENT_ACTIVATE = 'CV_ELEMENT_ACTIVATE';
const CV_STYLE_UPDATE = 'CV_STYLE_UPDATE';
const CV_TEXT_UPDATE = 'CV_TEXT_UPDATE';
const CV_BLOCK_SEND_BACK = 'CV_BLOCK_SEND_BACK';
const CV_BLOCKS_ALIGN_TOP = 'CV_BLOCKS_ALIGN_TOP';
const CV_BLOCKS_ALIGN_BOTTOM = 'CV_BLOCKS_ALIGN_BOTTOM';
const CV_BLOCKS_ALIGN_LEFT = 'CV_BLOCKS_ALIGN_LEFT';
const CV_BLOCKS_ALIGN_RIGHT = 'CV_BLOCKS_ALIGN_RIGHT';
const CV_BLOCKS_ALIGN_VERTICAL = 'CV_BLOCKS_ALIGN_VERTICAL';
const CV_BLOCKS_ALIGN_HORISONTAL = 'CV_BLOCKS_ALIGN_HORISONTAL';
const CV_BLOCKS_DISTRIBUTE_VERTICAL = 'CV_BLOCKS_DISTRIBUTE_VERTICAL';
const CV_BLOCKS_DISTRIBUTE_HORISONTAL = 'CV_BLOCKS_DISTRIBUTE_HORISONTAL';
const CV_BLOCKS_ALIGN_WIDTH = 'CV_BLOCKS_ALIGN_WIDTH';
const CV_BLOCKS_ALIGN_HEIGHT = 'CV_BLOCKS_ALIGN_HEIGHT';
const CV_BLOCKS_GROUP = 'CV_BLOCKS_GROUP';
const CV_BLOCK_UNGROUP = 'CV_BLOCK_UNGROUP';
const CV_BLOCK_LOCK = 'CV_BLOCK_LOCK';
const CV_LOAD = 'CV_LOAD';
const CV_INIT = 'CV_INIT';
const CV_SET_USER = 'CV_SET_USER';
const CV_SET_LINK = 'CV_SET_LINK';
const CV_BLOCK_LINK = 'CV_BLOCK_LINK';
const TEMPLATE_ADD = 'TEMPLATE_ADD';

const cvBlock_add = function(block) {
    return {
        type: CV_BLOCK_ADD,
        block:block,
    };
}

const cvBlock_delete = function(blockId) {
    return {
        type: CV_BLOCK_DELETE,
        blockId:blockId,
    };
}

const cvBlock_move = function(blockId,shiftTop,shiftLeft) {
    return {
        type: CV_BLOCK_MOVE,
        blockId:blockId,
        shiftTop: shiftTop,
        shiftLeft: shiftLeft,
    };
}

const cvBlock_resize = function(blockId,shiftHeight,shiftWidth) {
    return {
        type: CV_BLOCK_RESIZE,
        blockId:blockId,
        shiftHeight: shiftHeight,
        shiftWidth: shiftWidth,
    };
}

const cvBlock_activate = function(blockId,target) {
    return {
        type: CV_BLOCK_ACTIVATE,
        blockId:blockId,
        target:target,
    };
}

const cvElement_activate = function(style,elementId) {
    return {
        type: CV_ELEMENT_ACTIVATE,
        elementId:elementId,
        style:style,
    };
}

const cvStyle_update = function(blockId,styleName,styleValue) {
    return {
        type: CV_STYLE_UPDATE,
        blockId: blockId,
        styleName: styleName,
        styleValue: styleValue,
    };
}

const cvElement_textUpdate = function(blockId,textValue) {
    return {
        type: CV_TEXT_UPDATE,
        blockId: blockId,
        textValue: textValue,
    };
}

const cvBlock_sendBack = function(blockId) {
    return {
        type: CV_BLOCK_SEND_BACK,
        blockId: blockId,
    };
}

const cvBlock_copy = function(blockId) {
    return {
        type: CV_BLOCK_COPY,
        blockId: blockId,
    };
}

const cvBlock_setSize = function(blockId,height,width) {
    return {
        type: CV_BLOCK_SET_SIZE,
        blockId: blockId,
        width:width,
        height:height,
    };
}

const cvBlock_activateMulti = function(blockId) {
    return {
        type: CV_BLOCK_ACTIVATE_MULTI,
        blockId: blockId,
    };
}

const cvBlocks_align = function(mode) {
    switch (mode) {
        case 'top':
            return { type: CV_BLOCKS_ALIGN_TOP };
        case 'bottom':
            return { type: CV_BLOCKS_ALIGN_BOTTOM };
        case 'left':
            return { type: CV_BLOCKS_ALIGN_LEFT };
        case 'right':
            return { type: CV_BLOCKS_ALIGN_RIGHT };
        case 'vertical':
            return { type: CV_BLOCKS_ALIGN_VERTICAL };
        case 'horisontal':
            return { type: CV_BLOCKS_ALIGN_HORISONTAL };
        default:
    }
}

const cvBlocks_distribute = function(mode) {
    switch (mode) {
        case 'vertical':
            return { type: CV_BLOCKS_DISTRIBUTE_VERTICAL };
        case 'horisontal':
            return { type: CV_BLOCKS_DISTRIBUTE_HORISONTAL };
        default:
    }
}

const cvBlocks_alignSize = function(mode) {
    switch (mode) {
        case 'width':
            return { type: CV_BLOCKS_ALIGN_WIDTH };
        case 'height':
            return { type: CV_BLOCKS_ALIGN_HEIGHT };
        default:
    }
}

const cvBlocks_group = function() {
    return { type: CV_BLOCKS_GROUP };
}

const cvBlock_ungroup = function(blockId) {
    return { 
        type: CV_BLOCK_UNGROUP,
        blockId: blockId,
    };
}

const cvBlock_lock = function(blockId) {
    return { 
        type: CV_BLOCK_LOCK,
        blockId: blockId,
    };
}

const cv_load = function(blocks,style) {
    return { 
        type: CV_LOAD,
        blocks:blocks,
        style:style,
    };
}

const cv_init = function(blocks,style,user,link,templatesData,templatesUser) {
    return { 
        type: CV_INIT,
        blocks:blocks,
        style:style,
        user:user,
        link:link,
        templatesData:templatesData,
        templatesUser:templatesUser,
    };
}

const cv_setUser = function(user) {
    return { 
        type: CV_SET_USER,
        user:user,
    };
}

const cv_setLink = function(link) {
    return { 
        type: CV_SET_LINK,
        link:link,
    };
}

const cvBlock_setLink = function(blockId,linkValue) {
    return { 
        type: CV_BLOCK_LINK,
        blockId:blockId,
        linkValue:linkValue,
    };
}

const templates_add = function(blockId) {
    return {
        type: TEMPLATE_ADD,
        blockId: blockId,
    };
}

export {
    cvBlock_add, CV_BLOCK_ADD,
    cvBlock_delete, CV_BLOCK_DELETE,
    cvBlock_move, CV_BLOCK_MOVE,
    cvBlock_resize, CV_BLOCK_RESIZE,
    cvBlock_activate, CV_BLOCK_ACTIVATE,
    cvBlock_activateMulti, CV_BLOCK_ACTIVATE_MULTI,
    cvBlock_sendBack, CV_BLOCK_SEND_BACK,
    cvBlock_copy, CV_BLOCK_COPY,
    cvBlock_setSize, CV_BLOCK_SET_SIZE,
    cvBlock_lock, CV_BLOCK_LOCK,
    cvBlock_setLink, CV_BLOCK_LINK,
    cvBlocks_align, CV_BLOCKS_ALIGN_TOP, CV_BLOCKS_ALIGN_BOTTOM, CV_BLOCKS_ALIGN_LEFT, CV_BLOCKS_ALIGN_RIGHT, CV_BLOCKS_ALIGN_VERTICAL, CV_BLOCKS_ALIGN_HORISONTAL,
    cvBlocks_distribute, CV_BLOCKS_DISTRIBUTE_VERTICAL, CV_BLOCKS_DISTRIBUTE_HORISONTAL,
    cvBlocks_alignSize, CV_BLOCKS_ALIGN_WIDTH, CV_BLOCKS_ALIGN_HEIGHT,
    cvBlocks_group, CV_BLOCKS_GROUP, 
    cvBlock_ungroup, CV_BLOCK_UNGROUP,
    cvElement_activate, CV_ELEMENT_ACTIVATE,
    cvStyle_update, CV_STYLE_UPDATE,
    cvElement_textUpdate, CV_TEXT_UPDATE,
    cv_load, CV_LOAD,
    cv_init, CV_INIT, 
    cv_setUser, CV_SET_USER,
    cv_setLink, CV_SET_LINK,
    templates_add, TEMPLATE_ADD
}
