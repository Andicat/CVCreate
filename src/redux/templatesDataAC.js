const TEMPLATE_LOAD = 'TEMPLATE_LOAD';

const templates_load = function(data) {
    return {
        type: TEMPLATE_LOAD,
        data: data,
    };
}

export { templates_load, TEMPLATE_LOAD }
