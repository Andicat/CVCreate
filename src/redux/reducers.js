import { combineReducers } from 'redux';

import cvDataReducer from "./cvDataReducer";

let combinedReducer=combineReducers({
    cvData: cvDataReducer,
});

export default combinedReducer;
