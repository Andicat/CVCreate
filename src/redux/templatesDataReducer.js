import { TEMPLATE_LOAD } from './templatesDataAC';

const initState = {
    templatesArr: [],
    imageUrl: null,
}

function templatesDataReducer(state = initState, action) {

    switch (action.type) {
        //load
        case TEMPLATE_LOAD: {
            let newState = {templatesArr:action.data.templates, imageUrl:action.data.image};
            return newState;
        }
        default:
            return state;
    }
}

export default templatesDataReducer;
