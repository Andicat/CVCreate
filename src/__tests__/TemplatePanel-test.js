"use strict";

import {renderComponentWithStore, storeWithData} from './utils-test';
import {cvBlock_add} from '../redux/cvDataAC';

import TemplatePanel from '../components/TemplatePanel';

let component;

describe('TemplatePanel render', () => {

    it('Mеню шаблонов', () => {
        component = renderComponentWithStore(TemplatePanel,storeWithData);
        expect(component.toJSON()).toMatchSnapshot();
        //откроем группу шаблонов
        let templateGroup = component.root.findAll( el => el.props.className=='template-panel__group-name')[0];
        templateGroup.props.onClick();
    });

    it('Mеню шаблонов - Открытие группы', () => {
        component = renderComponentWithStore(TemplatePanel,storeWithData);
        let templateGroup = component.root.findAll( el => el.props.className=='template-panel__group-name')[0];
        templateGroup.props.onClick();
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe('TemplatePanel actions', () => {

    it('Добавление шаблона на страницу', () => {
        storeWithData.clearActions();
        component = renderComponentWithStore(TemplatePanel,storeWithData);
        let templateGroup = component.root.findAll( el => el.props.className=='template-panel__group-name')[0];
        templateGroup.props.onClick();
        let templateAdd = component.root.findAll( el => el.props.className=='template-panel__block-add')[0];
        const evt = {"target":{"previousSibling":document.createElement('div')}};
        templateAdd.props.onClick(evt);
        const expectedActions = [
            cvBlock_add({...storeWithData.getState().cvData.templates[0].elements[0],width:0,height:0}),
        ];
        expect(storeWithData.getActions()).toEqual(expectedActions);
    });
});
