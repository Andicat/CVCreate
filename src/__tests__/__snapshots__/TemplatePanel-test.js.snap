// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplatePanel render Mеню шаблонов - Открытие группы 1`] = `
<ul
  className="template-panel__menu undefined"
>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name template-panel__group-name--active"
      onClick={[Function]}
    >
      Image
    </div>
    <ul>
      <li
        className="template-panel__block entering"
      >
        <div
          className="template-panel__block-view"
        >
          <img
            alt=""
            className="cv__element cv__element--image"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "file": "",
                "opacity": 1,
              }
            }
          />
        </div>
        <button
          className="template-panel__block-add"
          onClick={[Function]}
        />
      </li>
      <li
        className="template-panel__block entering"
      >
        <div
          className="template-panel__block-view"
        >
          <img
            alt=""
            className="cv__element cv__element--image"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "borderRadius": "50%",
                "file": "",
                "opacity": 1,
              }
            }
          />
        </div>
        <button
          className="template-panel__block-add"
          onClick={[Function]}
        />
      </li>
      <li
        className="template-panel__block entering"
      >
        <div
          className="template-panel__block-view"
        >
          <img
            alt=""
            className="cv__element cv__element--image"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "borderColor": "#E05B49",
                "borderStyle": "solid",
                "borderWidth": "3px",
                "file": "",
                "opacity": 1,
              }
            }
          />
        </div>
        <button
          className="template-panel__block-add"
          onClick={[Function]}
        />
      </li>
      <li
        className="template-panel__block entering"
      >
        <div
          className="template-panel__block-view"
        >
          <img
            alt=""
            className="cv__element cv__element--image"
            data-elem={true}
            onClick={[Function]}
            style={
              Object {
                "borderColor": "#E05B49",
                "borderRadius": "50%",
                "borderStyle": "solid",
                "borderWidth": "3px",
                "file": "",
                "opacity": 1,
              }
            }
          />
        </div>
        <button
          className="template-panel__block-add"
          onClick={[Function]}
        />
      </li>
    </ul>
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Text
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Info block
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Figure
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Progress
    </div>
    <ul />
  </li>
</ul>
`;

exports[`TemplatePanel render Mеню шаблонов 1`] = `
<ul
  className="template-panel__menu undefined"
>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Image
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Text
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Info block
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Figure
    </div>
    <ul />
  </li>
  <li
    className="template-panel__group"
  >
    <div
      className="template-panel__group-name"
      onClick={[Function]}
    >
      Progress
    </div>
    <ul />
  </li>
</ul>
`;
